components/report/ReportModal.vue

```diff
@@ -0,0 +1,174 @@
+<script setup lang="ts">
+import type { mastodon } from 'masto'
+import { toggleBlockAccount, toggleFollowAccount, toggleMuteAccount, useRelationship } from '~~/composables/masto/relationship'
+
+const { account, status } = defineProps<{
+  account: mastodon.v1.Account
+  status: mastodon.v1.Status
+}>()
+
+const emit = defineEmits<{
+  (event: 'close'): void
+}>()
+
+const { client } = useMasto()
+
+const step = ref('choose')
+const serverRules = ref((await client.value.v2.instance.fetch()).rules || [])
+const reportReason = ref('')
+const selectedRuleIds = ref([])
+const additionalComments = ref('')
+const forwardReport = ref(false)
+
+const dismissButton = ref<HTMLDivElement>()
+
+async function submitReport() {
+  await client.value.v1.reports.create({
+    accountId: account.id,
+    statusIds: status?.id ? [status.id] : null,
+    comment: additionalComments.value,
+    forward: forwardReport.value,
+    category: reportReason.value === 'spam' ? 'spam' : reportReason.value === 'violation' ? 'violation' : 'other',
+    ruleIds: reportReason.value === 'violation' ? selectedRuleIds.value : null,
+  })
+  step.value = 'thanks'
+  // TODO: extract this scroll/reset logic into ModalDialog element
+  dismissButton.value?.scrollIntoView() // scroll to top
+}
+
+function unfollow() {
+  emit('close')
+  toggleFollowAccount(useRelationship(account).value!, account)
+}
+
+function mute() {
+  emit('close')
+  toggleMuteAccount(useRelationship(account).value!, account)
+}
+
+function block() {
+  emit('close')
+  toggleBlockAccount(useRelationship(account).value!, account)
+}
+</script>
+
+<template>
+  <div my-8 px-3 sm:px-8 flex="~ col gap-4" relative>
+    <h2 mxa text-xl>
+      {{ reportReason === 'dontlike' ? "Limiting" : "Reporting" }} <b text-primary>@{{ account.acct }}</b>
+    </h2>
+    <button ref="dismissButton" btn-action-icon absolute top--8 right-0 m1 aria-label="Close" @click="emit('close')">
+      <div i-ri:close-line />
+    </button>
+
+    <template v-if="step === 'choose'">
+      <h1 mxa text-4xl mb4>
+        Tell us what's wrong with this {{ status ? "post" : "account" }}
+      </h1>
+      <p text-xl>
+        Choose the best match:
```

Comment:
what about localizing these strings?

Reply:
I was planning to localize these once the rest of the team has tested on staging and we are all aligned on the flow. What do you think?

Reply:
Made a JIRA sub-task for this: https://mozilla-hub.atlassian.net/browse/MZTDN-184